src/SMU-Agilent_B29xx/main.py

```diff
@@ -390,16 +399,43 @@ def apply(self):
         value = str("{:.4E}".format(self.value))  # makes sure that self.value fits into SCPI command in terms of length
         if self.pulse:
             # get channel ready at specified values
+ 
             self.port.write(":SOUR%s:%s %s" % (self.channel, self.commands[self.source], self.pulseofflevel))
-            # arming the pulse trigger
-            self.port.write(":SOUR%s:%s:TRIG %s" % (self.channel, self.commands[self.source], value))
+    
+            # arming the pulse trigger; mis-using the sweep function with identical start and stop values so we can use number-of-pulses functionality
+            if self.source.startswith("Current"):
+                self.port.write(":SOUR%s:CURR:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:CURR:STOP %s" % (self.channel, value))
+            
+            if self.source.startswith("Voltage"):
+                self.port.write(":SOUR%s:VOLT:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:VOLT:STOP %s" % (self.channel, value))
+            
             # releasing the pulse trigger
             self.port.write(":INIT (@%s)" % self.channel)
+            
         else:
             # set output to specified values
             self.port.write(":SOUR%s:%s %s" % (self.channel, self.commands[self.source], value))
 
     def call(self):
+        
+        if self.pulse:
+            opcounter = 0                                                    # set counter for operation register request loop back to zero
+                
+            while True:
+                self.port.write(":STAT:OPER:COND?")                           # query SMU for the status of the operation register
+                opstatus=self.port.read()
+                print("Operation Status:", opstatus)
+                if opstatus=="1170":                                          # checks whether transition and aquisition are finished on both channels (INT values 2+16 + 128+1024)
```

Comment:
Could it be that 1170 is achieved in your case, but other users where other operation status values might be added as well would have a different total number here. Would it make sense to check whether single bits have the correct state?

Reply:
Giving this a second though, it could well be than one is using channel A for a constant voltage and channel B in a consecutive SweepMe step to emit a pulse. In this case, transistion and aqusition on channel A would not be completed.

Proposal: check which channel the procedure is running on and check only for >=18 (CH1) and >=1152 (CH2)

Reply:
I did not read into how the operation status is organized, but yes it would make sense to do it for each selected channel individually and to check only those bits that contain relevant information.